src/tokenizer.rs

```diff
@@ -454,9 +465,11 @@ impl<'a> Tokenizer<'a> {
         self.position += len_utf8;
         // Note that due to the special case for the 4-byte sequence
         // intro, we must use wrapping add here.
+        let len_utf16 = c.len_utf16();
         self.current_line_start_position = self
             .current_line_start_position
-            .wrapping_add(len_utf8 - c.len_utf16());
+            .wrapping_add(len_utf8 - len_utf16);
```

Comment:
I wonder if we can just store the utf-16 to utf-8 difference, then your current_position is just `position + difference`, right? You need to store the difference at the beginning of the line too, but it seems you could make those two `u16`s in practice and avoid growing the parser state, plus it'd be less work.

Reply:
I'll try that

> but it seems you could make those two u16s …

when you say those two, it's `difference` and `position`, right

> … in practice and avoid growing the parser state

I'm not following I think. I'd still need to have `difference` stored somewhere right? or do you mean that by just having 2 u16 instead of the 2 usize I have in my patch, we would end up not changing the parser state size?

Reply:
> I'll try that
> 
> > but it seems you could make those two u16s …
> 
> when you say those two, it's `difference` and `position`, right

No, I mean `difference` and `start_line_difference` or something along those lines.

> I'm not following I think. I'd still need to have `difference` stored somewhere right? or do you mean that by just having 2 u16 instead of the 2 usize I have in my patch, we would end up not changing the parser state size?

Yeah, I think you can avoid `current_position` if you keep `difference` and `start_line_difference`. Both of those can be u16, and the number of operations you do during parsing is less, because you only keep `difference` up-to-date, then `start_line_difference` is just an assignment when you parse a new line.

Reply:
I tried to do what you were suggesting, which seems to be working, but I'm still growing the parser state, so not sure I understood everything